frontend/src/components/authentication/authenticationGuard.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { ComponentType } from "react";
+import { withAuthenticationRequired } from "@auth0/auth0-react";
+import { Spinner } from '@chakra-ui/react'
+
+interface ComponentProps {
+    componentAsProps: ComponentType<object>
+}
```

Comment:
Give the interface a more specific name (i.e. `IAuthenticationGuard` or `AuthenticationGuardProps`)